src/reducers/sourceTargetReducer.ts

```diff
@@ -20,18 +20,21 @@ import { ChainDetails, SourceTargetAction, SourceTargetState } from '../types/so
 export default function sourceTargetReducer(state: SourceTargetState, action: SourceTargetAction): SourceTargetState {
   switch (action.type) {
     case SourceTargetActionsTypes.SWAP_CHAINS: {
-      return {
-        [ChainDetails.SOURCE]: {
-          sourceApiConnection: state[ChainDetails.TARGET].targetApiConnection,
-          sourceChain: state[ChainDetails.TARGET].targetChain,
-          sourcePolkadotjsUrl: state[ChainDetails.TARGET].targetPolkadotjsUrl
-        },
-        [ChainDetails.TARGET]: {
-          targetApiConnection: state[ChainDetails.SOURCE].sourceApiConnection,
-          targetChain: state[ChainDetails.SOURCE].sourceChain,
-          targetPolkadotjsUrl: state[ChainDetails.SOURCE].sourcePolkadotjsUrl
-        }
-      };
+      if (action.payload!.chain !== state[ChainDetails.SOURCE].sourceChain) {
+        return {
+          [ChainDetails.SOURCE]: {
+            sourceApiConnection: state[ChainDetails.TARGET].targetApiConnection,
+            sourceChain: state[ChainDetails.TARGET].targetChain,
+            sourcePolkadotjsUrl: state[ChainDetails.TARGET].targetPolkadotjsUrl
+          },
+          [ChainDetails.TARGET]: {
+            targetApiConnection: state[ChainDetails.SOURCE].sourceApiConnection,
+            targetChain: state[ChainDetails.SOURCE].sourceChain,
+            targetPolkadotjsUrl: state[ChainDetails.SOURCE].sourcePolkadotjsUrl
+          }
+        };
+      }
+      return state;
```

Comment:
if not then what? Im not sure that a switch is needed here

Reply:
The logic behind this if is to validate if the chain that comes from the payload is different than the current chain. If this is the case then source and target values are flipped ( because an account from the target chain was selected ).
In case an account from the source is chosen then no changes are needed and the source/target remains the same.

Regarding the usage of the switch to determine the action is a pattern which mainly comes from redux reducers: https://reactjs.org/docs/hooks-reference.html#usereducer

At this point in time it does not makes too much sense but as soon as we need new actions on this reducer, the switch statement will have more sense.